x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisFailureIT.java

```diff
@@ -160,16 +160,28 @@ public void testFailsOnChecksumMismatch() {
         blobStore.setDisruption(new Disruption() {
             @Override
             public byte[] onRead(byte[] actualContents, long position, long length) {
-                final byte[] disruptedContents = actualContents == null ? null : Arrays.copyOf(actualContents, actualContents.length);
-                if (actualContents != null && countDown.countDown()) {
-                    // CRC32 should always detect a single bit flip
-                    disruptedContents[Math.toIntExact(position + randomLongBetween(0, length - 1))] ^= (byte) (1 << between(0, 7));
```

Comment:
I did not reproduce the failure so that this is a best-effort attempt to fix. It seems plausible that in edge cases the `countDown` may not reach to 0 (hence no disruption) if there are many null `actualContents` or it happens towards the end of total read requests.

Reply:
Why `actualContents` might be null?

Reply:
I just noticed it can be `null` but I am actually not sure why. I guess it might related to early read (read before write completes)? @DaveCTurner could you please help out here?

I don't think length can be `0`. Otherwise we will see assertion error from `randomLongBetween(0, length - 1)`.